pytest/lib/mocknet.py

```diff
@@ -50,30 +65,51 @@
 '''
 
 
-# set prefix = 'sharded-' to access sharded mocknet nodes
-def get_node(i, prefix=''):
-    n = GCloudNode(f'{prefix}{NODE_BASE_NAME}{i}')
-    n.machine.username = NODE_USERNAME
-    n.machine.ssh_key_path = NODE_SSH_KEY_PATH
+def get_node(hostname):
+    instance_name = hostname
+    n = GCloudNode(instance_name,
+                   username=NODE_USERNAME,
+                   project=PROJECT,
+                   ssh_key_path=NODE_SSH_KEY_PATH)
     return n
 
 
-def get_nodes(prefix=''):
-    return pmap(lambda i: get_node(i, prefix=prefix), range(NUM_NODES))
+def get_nodes():
+    machines = gcloud.list(project=PROJECT,
+                           username=NODE_USERNAME,
+                           ssh_key_path=NODE_SSH_KEY_PATH)
+    nodes = []
+    pmap(
+        lambda machine: nodes.append(
+            GCloudNode(machine.name,
+                       username=NODE_USERNAME,
+                       project=PROJECT,
+                       ssh_key_path=NODE_SSH_KEY_PATH)), machines)
+    return nodes
+
+
+def node_account_name(node):
+    return f'{node.instance_name}.near'
 
 
-def create_target_dir(machine):
+def load_testing_account_id(node_account_id, i):
+    return f'load_testing_{i}_{node_account_id}'
+
+
+def create_target_dir(node):
     base_target_dir = os.environ.get(KEY_TARGET_ENV_VAR, DEFAULT_KEY_TARGET)
-    target_dir = f'{base_target_dir}/{machine.name}'
+    target_dir = f'{base_target_dir}/{node.instance_name}'
     run(f'mkdir -p {target_dir}')
     return target_dir
 
 
 def get_validator_account(node):
-    m = node.machine
-    target_dir = create_target_dir(m)
-    m.download(f'/home/ubuntu/.near/validator_key.json', target_dir)
-    return Key.from_json_file(f'{target_dir}/validator_key.json')
+    validator_key_file = tempfile.NamedTemporaryFile(
+        mode='r+', delete=False, suffix=f'.mocknet.loadtest.validator_key')
+    validator_key_file.close()
+    node.machine.download(f'/home/ubuntu/.near/validator_key.json',
+                          validator_key_file.name)
+    return Key.from_json_file(validator_key_file.name)
```

Comment:
Even if you don’t use the file returned by `NamedTemporaryFile` you
should still be able to do:

    validator_key_file = tempfile.NamedTemporaryFile(
        mode='r+', suffix=f'.mocknet.loadtest.validator_key')
        node.machine.download(f'/home/ubuntu/.near/validator_key.json',
                              validator_key_file.name)
        return Key.from_json_file(validator_key_file.name)

which will properly clean up the temporary file at the end.  Same in
other places.